inc/namespace.php

```diff
@@ -1151,6 +1187,19 @@ function elasticpress_mapping( array $mapping, ?string $index = null ) : array {
 	return $mapping;
 }
 
+/**
+ * Returns where inline settings should be used.
+ *
+ * @return bool
+ */
+function should_inline_settings() : bool {
+	$es_version = Elasticsearch::factory()->get_elasticsearch_version();
+
+	return version_compare( $es_version, '7.4', '>=' )
```

Comment:
Can we add a comment on why it's version 7.4 here? e.g.
```suggestion
	// AWS Elasticsearch 7.4+ is required for the _open and _close API to update index setting.
	// See https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-operations.html#version_api_notes
	return version_compare( $es_version, '7.4', '>=' )
```

Reply:
@roborourke I haven't actually touched the index update logic, does it happen behind the scenes ? or did I miss a whole block of functionality to cover ?

Reply:
@roborourke Also, the 7.4 check is for the user dictionary changes, which only came in effect after 7.4.

Reply:
This is more like the actual reason:

```suggestion
	// Elasticsearch 7.4 is where the inline user_dictionary_rules config option was introduced.
	return version_compare( $es_version, '7.4', '>=' )
```